src/libsyntax/parse/parser.rs

```diff
@@ -5114,7 +5120,39 @@ impl<'a> Parser<'a> {
                     if has_parens {
                         self.expect(&token::CloseDelim(token::Paren))?;
                     }
-                    let poly_trait = PolyTraitRef::new(lifetime_defs, path, lo.to(self.prev_span));
+                    let poly_span = lo.to(self.prev_span);
+                    if is_negative {
+                        let mut err = self
+                            .struct_span_err(poly_span,
+                                             "negative trait bounds are not supported");
+                        if colon_span.is_some() {
+                            err.span_suggestion_with_applicability(
+                                last_plus_span.or_else(|| {
+                                    has_removed_colon = true;
+                                    colon_span
+                                }).unwrap().to(poly_span),
+                                "remove this trait bound",
+                                "".into(),
+                                Applicability::MachineApplicable);
+                        } else {
+                            err.help("remove this trait bound");
```

Comment:
If the other comment is implemented, this is the branch where you would have to account for the trailing `+` span instead, and suggest the removal of it, instead of in the warning.